integrated_channels/canvas/client.py

```diff
@@ -378,37 +378,56 @@ def _handle_canvas_assignment_retrieval(
             course_id (str) : the Canvas course ID relating to the course which the client is currently
             transmitting learner data to.
         """
-        # First, check if the course assignment already exists
+        # Check if the course assignment already exists
         canvas_assignments_url = '{canvas_base_url}/api/v1/courses/{course_id}/assignments'.format(
             canvas_base_url=self.enterprise_configuration.canvas_base_url,
             course_id=course_id
         )
         resp = self.session.get(canvas_assignments_url)
 
-        if resp.status_code >= 400:
-            raise ClientError(
-                "Something went wrong retrieving assignments from Canvas. Got response: {}".format(
-                    resp.text,
-                ),
-                resp.status_code
-            )
+        more_pages_present = True
+        assignment_id = ''
 
-        assignments_resp = resp.json()
-        assignment_id = None
-        for assignment in assignments_resp:
-            try:
-                if assignment['integration_id'] == integration_id:
-                    assignment_id = assignment['id']
-                    break
-            # The validation check above should ensure that we have a 200 response from Canvas, but sanity catch if we
-            # have a unexpected response format
-            except (KeyError, ValueError, TypeError) as error:
+        while more_pages_present:
+            if resp.status_code >= 400:
                 raise ClientError(
                     "Something went wrong retrieving assignments from Canvas. Got response: {}".format(
                         resp.text,
                     ),
                     resp.status_code
-                ) from error
+                )
+
+            assignments_resp = resp.json()
+            for assignment in assignments_resp:
+                try:
+                    if assignment['integration_id'] == integration_id:
+                        assignment_id = assignment['id']
+                        break
+
+                # The validation check above should ensure that we have a 200 response from Canvas,
+                # but sanity catch if we have a unexpected response format
+                except (KeyError, ValueError, TypeError) as error:
+                    raise ClientError(
+                        "Something went wrong retrieving assignments from Canvas. Got response: {}".format(
+                            resp.text,
+                        ),
+                        resp.status_code
+                    ) from error
+
+            if not assignment_id:
+                # Canvas pagination headers come back as a string,
+                # so we have to parse out the linked list of assignment pages
```

Comment:
This parsing is hard to read as a reader of code, please add a response example or a URL link to canvas docs for why you know this parser is correct.